Makefile

```diff
@@ -1,15 +1,36 @@
+OS_FLAVOR ?= origin
+# Use for e.g. --cache-only
+COMPOSEFLAGS ?=
+ROOT_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+CACHE_ARGS := $(shell if test -d cache; then echo $(shell pwd)/cache; fi)
+
+all: rpmostree-compose
+
 .PHONY: syntax-check
 syntax-check:
-	@set -e; for jsonfile in $$(find . -name '*.json'); do \
+	@set -e; for jsonfile in $$(find ${ROOT_DIR} -name '*.json'); do \
 		echo -n "Checking JSON syntax for $${jsonfile}... "; \
 		jq < $${jsonfile} . >/dev/null; \
 		echo "OK"; \
 	done
 
 .PHONY: container
 container: repo-refresh
-	imagebuild -privileged .
+	imagebuild -privileged ${ROOT_DIR}
 
 .PHONY: repo-refresh
 repo-refresh:
-	./generate-openshift-repo
+	${ROOT_DIR}/generate-openshift-repo
+
+.PHONY: init-ostree-repo
+init-ostree-repo:
+	ostree --repo=build-repo init --mode=bare-user
+	ostree --repo=repo init --mode=archive
+
+.PHONY: rpmostree-compose
+rpmostree-compose: ${ROOT_DIR}/openshift.repo init-ostree-repo
+	if test -d cache; then cachedir='--cachedir $(shell pwd)/cache'; fi && \
+	  cd ${ROOT_DIR} && set -x && \
+	  rpm-ostree $(COMPOSEFLAGS) compose tree $${cachedir:-} --repo=$(shell pwd)/build-repo host-$(OS_FLAVOR).json
```

Comment:
1) `$$` quotes to `$` for the shell (`make` also uses `$`).
2) So I can pass a variable.  Otherwise using `\` etc. is annoying.
3) No caching of that form in `Dockerfile` builds today.